src/jshint.js

```diff
@@ -2508,7 +2508,11 @@ var JSHINT = (function() {
           // operator.
           (isFunctor(ret) && !isEndOfExpr()) ||
           // Used as the return value of a single-statement arrow function
-          (ret.id === "{" && preceeding.id === "=>");
+          (ret.id === "{" && preceeding.id === "=>") ||
+          // Used to delineate an integer number literal from a dereferencing
+          // punctuator (otherwise interpreted as a decimal point)
+          (ret.type === "(number)" &&
+            checkPunctuators(pn, ".") && /^\d+$/.test(ret.value));
```

Comment:
it should be checkPunctuators(pn,  ["."]) for consistency shouldn't it? at least till my pr is merged